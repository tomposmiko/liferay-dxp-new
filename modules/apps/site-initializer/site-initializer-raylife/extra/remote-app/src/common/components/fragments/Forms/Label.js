import React from 'react';

export const Label = ({children, label, name, required = false}) => {
	return (
		<label htmlFor={name}>
			<span className={`${required && 'required'}`}>{label}</span>
			{children}
		</label>
	);
};
