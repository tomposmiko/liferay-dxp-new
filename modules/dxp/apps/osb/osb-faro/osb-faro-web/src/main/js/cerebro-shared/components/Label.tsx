import Button from 'shared/components/Button';
import getCN from 'classnames';
import React from 'react';

interface ILabelProps extends React.HTMLAttributes<HTMLSpanElement> {
	className?: string;
	closeable: boolean;
	display: 'danger' | 'info' | 'secondary' | 'success' | 'warning';
	label: string;
	onRemove?: () => void;
	size?: 'lg';
}

const Label: React.FC<ILabelProps> = ({
	className,
	closeable = false,
	display = 'secondary',
	label,
	onRemove,
	size
}) => {
	const classes = getCN('label', `label-${display}`, className, {
		['label-dismissible']: closeable,
		[`label-${size}`]: size
	});

	return (
		<span className={classes}>
			<span className='label-item label-item-expand'>{label}</span>

			{closeable && (
				<span className='label-item label-item-after'>
					<Button
						aria-label={Liferay.Language.get('close')}
						className='close'
						display='unstyled'
						icon='times'
						iconAlignment='right'
						onClick={onRemove}
					/>
				</span>
			)}
		</span>
	);
};

export default Label;
