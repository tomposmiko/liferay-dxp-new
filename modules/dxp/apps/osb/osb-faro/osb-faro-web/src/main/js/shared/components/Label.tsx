import ClayButton from '@clayui/button';
import ClayIcon from '@clayui/icon';
import getCN from 'classnames';
import omitDefinedProps from 'shared/util/omitDefinedProps';
import React, {FC} from 'react';

enum Displays {
	Primary = 'primary',
	Secondary = 'secondary',
	Success = 'success',
	Info = 'info',
	Warning = 'warning',
	Danger = 'danger',
	Light = 'light',
	Dark = 'dark'
}

enum Sizes {
	Large = 'lg'
}

const Label: FC<{
	className?: string;
	display?: string;
	index?: number;
	onRemove?: (index?: number) => void;
	size?: string;
	uppercase?: boolean;
}> & {Displays: typeof Displays; Sizes: typeof Sizes} = props => {
	const {
		children,
		className,
		display,
		index,
		onRemove,
		size,
		uppercase = false,
		...otherProps
	} = props;

	const handleRemove = () => {
		onRemove && onRemove(index);
	};

	const classes = getCN('label', 'label-root', className, {
		[`label-${display}`]: display,
		[`label-${size}`]: size,
		'label-dismissible': onRemove,
		['label-uppercase']: uppercase
	});

	return (
		<div {...omitDefinedProps(otherProps, props)} className={classes}>
			<span className='label-item'>{children}</span>

			{onRemove && (
				<span className='label-item label-item-after'>
					<ClayButton
						aria-label={Liferay.Language.get('close')}
						className='button-root close'
						displayType='secondary'
						onClick={handleRemove}
					>
						<ClayIcon className='icon-root' symbol='times' />
					</ClayButton>
				</span>
			)}
		</div>
	);
};

Label.Displays = Displays;
Label.Sizes = Sizes;

export default Label;
