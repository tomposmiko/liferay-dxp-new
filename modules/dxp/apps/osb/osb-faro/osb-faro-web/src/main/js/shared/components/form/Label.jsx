import getCN from 'classnames';
import InfoPopover from 'shared/components/InfoPopover';
import omitDefinedProps from 'shared/util/omitDefinedProps';
import React from 'react';
import {PropTypes} from 'prop-types';

export default class Label extends React.Component {
	static defaultProps = {
		required: false
	};

	static propTypes = {
		className: PropTypes.string,
		popover: PropTypes.object,
		required: PropTypes.bool
	};

	render() {
		const {
			children,
			className,
			popover,
			required,
			...otherProps
		} = this.props;

		return (
			<label
				{...omitDefinedProps(otherProps, Label.propTypes)}
				className={getCN(
					'form-control-label',
					'label-root',
					className,
					{
						required
					}
				)}
			>
				<span className='content-container'>{children}</span>

				{popover && (
					<InfoPopover
						content={popover.content}
						title={popover.title}
					/>
				)}
			</label>
		);
	}
}
