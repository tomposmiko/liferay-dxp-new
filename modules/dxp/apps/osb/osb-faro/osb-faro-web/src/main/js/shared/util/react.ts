import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
