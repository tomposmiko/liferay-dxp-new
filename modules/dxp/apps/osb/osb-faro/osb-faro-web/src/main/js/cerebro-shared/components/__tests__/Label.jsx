import Label from '../Label';
import React from 'react';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('Label', () => {
	it('should render', () => {
		const {container} = render(<Label label='Label' />);

		expect(container).toMatchSnapshot();
	});

	it('should render closeable Label', () => {
		const {container} = render(<Label closeable />);

		expect(container.querySelector('.close')).toBeTruthy();
	});
});
