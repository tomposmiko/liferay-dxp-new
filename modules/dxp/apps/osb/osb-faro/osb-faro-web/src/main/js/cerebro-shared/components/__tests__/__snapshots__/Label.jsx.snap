// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render 1`] = `
<div>
  <span
    class="label label-secondary"
  >
    <span
      class="label-item label-item-expand"
    >
      Label
    </span>
  </span>
</div>
`;
